pkg/secret/replicated.go

```diff
@@ -161,10 +165,14 @@ func HasReplicatedSecrets(ctx context.Context, c client.Client, kcKey client.Obj
 
 func generateReplicatedSecret(kcKey client.ObjectKey, replicationTargets []replicationapi.ReplicationTarget) *replicationapi.ReplicatedSecret {
 	return &replicationapi.ReplicatedSecret{
-		ObjectMeta: getManagedObjectMeta(kcKey.Name, kcKey),
+		ObjectMeta: metav1.ObjectMeta{
+			Name:      kcKey.Name,
+			Namespace: kcKey.Namespace,
+			Labels:    labels.WatchedByK8ssandraClusterLabels(kcKey),
```

Comment:
Shouldn't we consider that a ReplicatedSecret is "part of" the K8ssandraCluster? Since it's been created by the controller, I'd set it part of, which will naturally add the WatchedBy labels.
I think WatchedBy should only be applied to externally provided resources.

Reply:
A lot of the other label sets are supersets of "watched by". So yes, we could document that they "imply" it, and avoid redundant declaration.
I'd like to address that after #860 if that's OK.

Reply:
sounds good